fs/parallel_artifact_fetcher.go

```diff
@@ -273,9 +278,25 @@ func (f *parallelArtifactFetcher) fetchFromRemoteAndWriteToTempDir(ctx context.C
 	if err != nil {
 		return nil, fmt.Errorf("error reopening file at %s after writing: %w", ingestPath, err)
 	}
+
+	// start async verification of the blob digest
+	if err = f.asyncVerifyBlobDigest(ctx, ingestPath); err != nil {
```

Comment:
This stuck out to me a little but I guess thinking about it we should verify the digest of every object we download so it makes sense to verify the digest here. Should we also do this in the non-parallel artifact fetcher? Or does this not work without having the file on disk?

Reply:
Ah ok, I forgot to comment on this before but we make this asynchronous by opening another file descriptor, by the time we get to the layerArchive we lose this ref. So it's probably best to just use our own fd here.